projboard/tests/test_client.py

```diff
@@ -23,32 +41,107 @@ def test_get_searchpage(client):
 
 
 @pytest.fixture
 @pytest.mark.django_db
-def search_by_title():
-    articles = Article.search_by_title(VALUE)
-    return articles
+def articles_by_title():
+    articles_by_title = Article.search_by_title(VALIDTITLE)
+    return articles_by_title
 
 
+@pytest.fixture
 @pytest.mark.django_db
-def test_valued_searchpage_results(client, articles):
-    response = client.post('/search/', {'title': VALUE})
+def invalid_articles_by_title():
+    invalid_articles_by_title = Article.search_by_title(INVALIDINPUT)
+    return invalid_articles_by_title
+
+
+@pytest.mark.django_db
+def test_valid_title_searchpage_results(client, articles_by_title):
+    response = client.post('/search/', {'searchInput': VALIDTITLE, 'searchOptions': 'title'})
     assert response.status_code == 200
-    assert response.context[SEARCHTITLE] == VALUE
+    assert response.context[SEARCHINPUT] == VALIDTITLE
     for i in response.context[ARTICLES]:
-        assert i in articles
-    assert response.context[COUNT] == len(articles)
+        assert i in articles_by_title
+    assert response.context[COUNT] == len(articles_by_title)
     assert response.context[MESSAGE] == ""
 
 
+@pytest.mark.django_db
+def test_invalid_title_searchpage_results(client, invalid_articles_by_title):
+    response = client.post('/search/', {'searchInput': INVALIDINPUT, 'searchOptions': 'title'})
+    assert response.status_code == 200
+    assert response.context[SEARCHINPUT] == INVALIDINPUT
+    for i in response.context[ARTICLES]:
+        assert i in invalid_articles_by_title
+    assert response.context[COUNT] == len(invalid_articles_by_title)
+    assert response.context[MESSAGE] == WRONG_TITLE_MESSAGE
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def articles_by_subject():
+    subject = Subject.objects.get(name=VALIDSUBJECT)
+    articles_by_subject = Article.search_by_subject(subject.id)
+    return articles_by_subject
+
+
+@pytest.mark.django_db
+def test_valid_subject_searchpage_results(client, articles_by_subject):
+    response = client.post('/search/', {'searchInput': VALIDSUBJECT, 'searchOptions': 'subject'})
+    assert response.status_code == 200
+    assert response.context[SEARCHINPUT] == VALIDSUBJECT
+    for i in response.context[ARTICLES]:
+        assert i in articles_by_subject
+    assert response.context[COUNT] == len(articles_by_subject)
+    assert response.context[MESSAGE] == ""
+
+
+@pytest.mark.django_db
+def test_invalid_subject_searchpage_result(client):
+    response = client.post('/search/', {'searchInput': INVALIDSUBJECT, 'searchOptions': 'subject'})
+    assert response.status_code == 200
+    assert response.context[SEARCHINPUT] == INVALIDSUBJECT
+    assert response.context[COUNT] == 0
+    assert response.context[MESSAGE] == WRONG_SUBJECT_MESSAGE
+
+
+@pytest.fixture
+@pytest.mark.django_db
+def articles_by_user(client):
+    user = User.get_user_by_nickname(VALIDUSER)
+    valid_articles_by_user = Article.search_by_user(user.id)
```

Comment:
Same here you can use a two fixtures that calling each other
and the fixture is holding the data to compare againsr